ClothesApp/Application/Services/CategoryService.cs

```diff
@@ -0,0 +1,190 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Application.Dtos.Category;
+using Application.Dtos.SectionCategories;
+using Application.Exceptions;
+using Application.Interfaces.Repositories;
+using Application.Interfaces.Services;
+using AutoMapper;
+using Domain.Entities;
+
+namespace Application.Services;
+
+public class CategoryService : ICategoryService
+{
+    private readonly ICategoryRepository _categoryRepository;
+
+    private readonly ISectionRepository _sectionRepository;
+
+    private readonly ISectionCategoryRepository _sectionCategoryRepository;
+
+    private readonly IMapper _mapper;
+
+    public CategoryService(ICategoryRepository categoryRepository, IMapper mapper, ISectionRepository sectionRepository, ISectionCategoryRepository sectionCategoryRepository)
+    {
+        _categoryRepository = categoryRepository;
+        _mapper = mapper;
+        _sectionRepository = sectionRepository;
+        _sectionCategoryRepository = sectionCategoryRepository;
+    }
+    
+    public async Task<CategoryDto> GetById(long id)
+    {
+        var category = await _categoryRepository.GetById(id);
+
+        if (category is null)
+        {
+            throw new NotFoundException(Messages.CategoryNotFound);
+        }
+
+        var categoryDto = _mapper.Map<CategoryDto>(category);
+
+        return categoryDto;
+    }
+
+    public async Task<string> BuildCategoryTree()
+    {       
+        JsonSerializerOptions options = new()
+        {
+            ReferenceHandler = ReferenceHandler.Preserve,
+            WriteIndented = true
+        };
+        
+        var categories = await _categoryRepository.GetAll();
```

Comment:
This looks quite complicated. And also I wouldn't create tree starting from the very top level, usually in estore you see top level sections, you click into one of them and you see list of top level categories.
And then when you click to category you see a tree of all subcategories because it would be stressful for user to go into 5-10 levels of depth
On the other hand showing one tree for the whole applications makes is confusing as well because there will be too many categories and nested levels
So I would suggest to have two endpoints: 1 for top level categories 2 for category tree by parent category id